src/cpp/client/client_context.cc

```diff
@@ -75,6 +76,10 @@ void ClientContext::set_call(grpc_call* call,
   }
 }
 
+std::unique_ptr<const AuthContext> ClientContext::auth_context() const {
+  return CreateAuthContext(call_);
```

Comment:
I think that CreateAuthContext should:
1. Return a `shared_ptr` as discussed above.
2. Be called from the `set_call` function so that a `shared_ptr` member variable can be set.

In this function (`auth_context()`), I think that you should just return this member variable. Makes sense?